atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/AbstractDbWriteTable.java

```diff
@@ -133,11 +135,25 @@ public void delete(List<Entry<Cell, Long>> entries) {
 
         String prefixedTableName = prefixedTableNames.get(tableRef);
         conns.get().updateManyUnregisteredQuery(" /* DELETE_ONE (" + prefixedTableName + ") */ "
-                + " DELETE /*+ INDEX(m pk_" + prefixedTableName + ") */ "
+                + " DELETE /*+ INDEX(m " + getPrimaryKeyConstraintName(prefixedTableName) + ") */ "
                 + " FROM " + prefixedTableName + " m "
                 + " WHERE m.row_name = ? "
                 + "  AND m.col_name = ? "
                 + "  AND m.ts = ?",
                 args);
     }
+
+    private String getPrimaryKeyConstraintName(String tableName) {
+        String pkConstraintName = "pk_" + tableName;
+        if (config.type().equals(OracleDdlConfig.TYPE)) {
+            return truncateToMaxOracleLength(pkConstraintName);
+        }
+        return pkConstraintName;
+    }
+
+    private String truncateToMaxOracleLength(String constraintName) {
+        return constraintName
+                .substring(0, Math.min(OracleTableNameMapper.ORACLE_MAX_TABLE_NAME_LENGTH, constraintName.length()));
```

Comment:
See above comment about making `getPrimaryKeyConstraintName` protected, but my take would be that this should throw an exception if the generated constraint name exceeds the DB's internal limitations. There have been some nasty issues in the past with silent truncation, so I'd like to avoid those. I realize that making this backwards compatible with previously generated tables & indexes is challenging, so I'd love to see specific tests around exact behavior for existing examples.